llvm-general/src/LLVM/General/Internal/FFI/FunctionC.cpp

```diff
@@ -55,8 +55,15 @@ void LLVM_General_SetFunctionCallingConvention(LLVMValueRef f, unsigned cc) {
   unwrap<Function>(f)->setCallingConv(llvm::CallingConv::ID(cc));
 }
 
-LLVMBool LLVM_General_HasPersonalityFn(LLVMValueRef Fn) {
-  return unwrap<Function>(Fn)->hasPersonalityFn();
+// This wrapper is necessary because LLVMSetPersonalityFn fails if
+// personalityFn is a nullptr even though the C++ API allows that.
+void LLVM_General_SetPersonalityFn(LLVMValueRef fn, LLVMValueRef personalityFn) {
+    llvm::Function* f = unwrap<Function>(fn);
+    if (personalityFn == nullptr) {
+        f->setPersonalityFn(nullptr);
+    } else {
+        f->setPersonalityFn(unwrap<Constant>(personalityFn));
+    }
```

Comment:
Might be more readable if you just use a ?: - i.e.

unwrap<Function>(fn)->setPersonalityFn(personalityFn ? unwrap<Constant>(personalityFn) : nullptr);